app/models/request.rb

```
@@ -112,13 +112,55 @@ def approve(canvas_facade, processed_user_id)
     return false unless response.success?
 
     assignment_override = JSON.parse(response.body)
+    send_email_response if course.course_settings&.enable_emails
     update(status: 'approved', last_processed_by_user_id: processed_user_id.id, external_extension_id: assignment_override['id'])
   end
 
   def reject(processed_user_id)
+    send_email_response if course.course_settings&.enable_emails
     update(status: 'denied', last_processed_by_user_id: processed_user_id.id)
   end
 
+  def generate_email_response
+    course_settings = course.course_settings
+    return 'Course settings not found.' unless course_settings
+
+    template = course_settings.email_template
+    subject = course_settings.email_subject
+
+    # Define the placeholders and their corresponding values
+    placeholders = {
+      '{{student_name}}' => user.name,
+      '{{assignment_name}}' => assignment.name,
+      '{{course_name}}' => course.course_name,
+      '{{course_code}}' => course.course_code,
+      '{{status}}' => status.capitalize,
+      '{{original_due_date}}' => assignment.due_date&.strftime('%a, %b %-d, %Y %-I:%M %p'),
+      '{{new_due_date}}' => requested_due_date&.strftime('%a, %b %-d, %Y %-I:%M %p'),
+      '{{extension_days}}' => calculate_days_difference.to_s
+    }
+
+    # Replace placeholders in the template using regex
+    placeholders.each do |placeholder, value|
+      subject = subject.gsub(placeholder, value || 'N/A')
+      template = template.gsub(placeholder, value || 'N/A')
+    end
+
+    { subject: subject, template: template }
+  end
+
+  def send_email_response
+    return unless course.course_settings&.enable_emails
+
+    course_email = course.course_settings.reply_email.presence || 'flextensions@berkeley.edu'
+    student_email = user.email
+    email_response = generate_email_response
```

Maybe validate the placeholders in the email template. To warn if the template contains placeholders that aren’t replaced or typos like `{{studnet_name}}` We might end up sending bad emails and have silent failures.